disnake/ext/commands/ctx_menus_core.py

```diff
@@ -73,15 +75,23 @@ def __init__(
         func: InteractionCommandCallback,
         *,
         name: str = None,
-        default_permission: bool = True,
+        default_permission: bool = MISSING,
+        dm_permission: bool = True,
+        default_member_permissions: Permissions = None,
         guild_ids: Sequence[int] = None,
         auto_sync: bool = True,
         **kwargs,
     ):
         super().__init__(func, name=name, **kwargs)
         self.guild_ids: Optional[Sequence[int]] = guild_ids
         self.auto_sync: bool = auto_sync
-        self.body = UserCommand(name=self.name, default_permission=default_permission)
+        self.body = UserCommand(
+            name=self.name,
+            default_permission=default_permission,
+            dm_permission=dm_permission and not self._guild_only,
```

Comment:
Yeah, would you add that?

Reply:
This isn't feasible as long as the `dm_permission` parameter defaults to `True` instead of a sentinel value, let's just keep it this way for now. Thinking about it again, there isn't really a need for a warning here, as it'll always go with the more restrictive option anyway, i.e. using `GuildCommandInteraction` and setting `dm_permission=True` will still result in `dm_permission=False`.